src/rest/CommunityEndpoints.js

```diff
@@ -28,87 +204,286 @@ export async function joinCommunity(communityAddress, memberAddress, secret = un
+/**
+ * Get stats of a single community member, including proof
+ * @param {EthereumAddress} communityAddress to query
+ * @param {EthereumAddress} memberAddress (optional) if not supplied, get the stats of currently logged in StreamrClient (if auth: privateKey)
+ */
+export async function getMemberStats(communityAddress, memberAddress) {
+    let address = memberAddress
+    if (!address) {
+        const authKey = this.options.auth && this.options.auth.privateKey
+        if (!authKey) {
+            throw new Error("StreamrClient wasn't authenticated with privateKey, and memberAddress argument not supplied")
+        }
+        address = computeAddress(authKey)
     }
-    return addr
+
+    return getOrThrow(this, communityAddress, `/members/${address}`)
 }
 
 /**
- * Deploy a new CommunityProduct contract and create the required joinPartStream
- * @param {Wallet} wallet to do the deployment from, also becomes owner or stream and contract
- * @param {Number} blockFreezePeriodSeconds security parameter against operator failure (optional, default: 0)
- * @param {Number} adminFee fraction of revenue that goes to product admin, 0...1 (optional, default: 0)
- * @param {Function} logger will print debug info if given (optional)
+ * @typedef {Object} BalanceResponse
+ * @property {BigNumber} total tokens earned less withdrawn previously, what you'd get once Operator commits the earnings to CommunityProduct contract
+ * @property {BigNumber} withdrawable number of tokens that you'd get if you withdraw now
  */
-async function deployCommunity(wallet, blockFreezePeriodSeconds = 0, adminFee = 0, logger) {
-    await throwIfNotContract(wallet.provider, this.options.tokenAddress, 'deployCommunity function argument tokenAddress')
-    await throwIfBadAddress(this.options.streamrNodeAddress, 'StreamrClient option streamrNodeAddress')
 
-    if (adminFee < 0 || adminFee > 1) { throw new Error('Admin fee must be a number between 0...1, got: ' + adminFee) }
-    const adminFeeBN = new utils.BigNumber((adminFee * 1e18).toFixed()) // last 2...3 decimals are going to be gibberish
+/**
+ * Calculate the amount of tokens the member would get from a successful withdraw
+ * @param communityAddress
+ * @param memberAddress
+ * @return {Promise<BalanceResponse>} earnings minus withdrawn tokens
+ */
+export async function getBalance(communityAddress, memberAddress, provider) {
+    let address = memberAddress
+    if (!address) {
+        const authKey = this.options.auth && this.options.auth.privateKey
+        if (!authKey) {
+            throw new Error("StreamrClient wasn't authenticated with privateKey, and memberAddress argument not supplied")
+        }
+        address = computeAddress(authKey)
+    }
 
-    const stream = await this.getOrCreateStream({
-        name: `Join-Part-${wallet.address.slice(0, 10)}-${Date.now()}`
+    const stats = await get(this, communityAddress, `/members/${address}`)
+    if (stats.error || stats.earnings === '0') {
```

Comment:
Note there's a suppressed error here that you might want to expose on the returned object somehow, especially if `stats.error` isn't an "expected" error. Otherwise user can't differentiate actual zero balances from infrastructure failure.

Reply:
True, but that one really is a convenience function. It says zero in case of failure; this may be spurious, true, but the "typical" stats.error should be member or community doesn't exist. Detecting infrastructure failure would be a good separate helper function maybe. Later, I won't start guessing right now what typical infra failure would look like.